### DISCUSSION TITLE
How to extract header embedded styles into separate css files in Nuxt3?

### DISCUSSION THREAD

**Discussion by Baiyuetribe** [nuxt/nuxt#16553] [Questions]
I have completed naive-ui to support nuxt3, there is an unimportant problem, that is, it is currently impossible to extract the header file as a separate css, please help https://github.com/Baiyuetribe/nuxt3-naive-ui-starter/issues/4
![](https://user-images.githubusercontent.com/43716063/154205964-178cfe08-0719-42c8-8108-c7f3e96538ff.png)

Mini DEOM: https://stackblitz.com/github/Baiyuetribe/nuxt3-naive-ui-starter

**Comment by smarroufin** [comment]
Hi @Baiyuetribe 
I invite you to post on nuxt/framework repository for any Nuxt 3 related question :slightly_smiling_face: 
@danielroe Could you transfer it please?

**Reply by Baiyuetribe** [reply]
At first I thought it was not important, until after I used multiple components, I found that the head accumulated a lot of CSS, which took up a lot of volume because it was not compressed.

**Reply by Baiyuetribe** [reply]
According to naive-ui maintainer feedback, these css styles will eventually be gzipped, so no need to worry about size.But, it would be better if it could be extracted into a separate css file

**Comment by lovefields** [comment]
Excuse me~

I want extract my scss to file too. Any one has solution or code?
